### PR TITLE
adds object/function filter to local plugin

### LINKED ISSUE #746: Advanced Filtering
Hey,

Like the project, jut wondering if it is possible to do more advanced filtering across multiple fields e.g. say with row data
`{ name: "Bob", age: 20, categories: ["male","young"] }, { name: "Alice", age: 49: categories: ["female", "middle-age"]`

Would it be possible to search by:

a) age between x and y
b) which categories a person is in
c) text search on name

From docs it looks like its just text across all of the properties? Are the more advanced searches that I am talking about possible?

Thanks,

Ben

### PR DESCRIPTION
**Pull request by dahlbyk**
Git fail. This all belongs to #747 to close #746.

### CODE REVIEW

File: stories/index.tsx [hunk 1]
```diff
@@ -1210,6 +1210,35 @@ storiesOf('Filter', module)
       </Griddle>
     )
   })
+  .add('with Custom Filter for the column "name"', () => {
+    class CustomFilter extends React.Component<{setFilter: (e: any) => any, style: any, className: any}, {}> {
+      public setFilter = (e: any) => {
+        console.log(e.target.value);
+        const filterAsMap = new Map();
+        filterAsMap.set('name', e.target.value);
+        this.props.setFilter(filterAsMap);
```
[the remark this answers is not in the document]
Comment (benoj) [reply stories/index.tsx]: Agreed. TBH I'm not a huge fan of Immutable.JS so would be good not to have that bleed into outside projects using Griddle

[the remark this answers is not in the document]
Comment (dahlbyk) [reply stories/index.tsx]: > would be good not to have that bleed into outside projects using Griddle

Agreed. I'm not thrilled with the places that it does leak, but it seems likely to be a nontrivial performance hit to un-Immutable `data` before exposing rows to filter functions (for example).

[the remark this answers is not in the document]
Comment (benoj) [reply stories/index.tsx]: Yes - This is probably the reason I left it there originally (It was a while back I started this...)

Might be safe to leave it as is for now for perf. reasons and look to add a feature where can optionally un-Immutable data before filtering?

[the remark this answers is not in the document]
Comment (benoj) [reply stories/index.tsx]: e.g. `setFilter(filter, { json: true })` where json means dont sent me immutableJs object

[the remark this answers is not in the document]
Comment (arnimhanke) [reply stories/index.tsx]: I agree with you two!
But I must admit that I don't know exactly which part of the code you want to be changed, so i cant give any input in this discussion :-/

[the remark this answers is not in the document]
Comment (dahlbyk) [reply stories/index.tsx]: https://github.com/GriddleGriddle/Griddle/pull/807/commits/7bacdb46e235d3715640788e2738ad58160aefec!

Added some TypeScript smarts, too.

[the remark this answers is not in the document]
Comment (arnimhanke) [reply stories/index.tsx]: As far as i can see it looks good :)
Is there something I can do for progress?

**Comment by dahlbyk** [comment]
I'll try to ship a point release with this yet today. Thanks @benoj and @arnimhanke!

**Comment by dahlbyk** [comment]
This has been released with `griddle-react@1.3.0`, though not tagged as `latest` yet.

cc @waclock

**Comment by arnimhanke** [comment]
Do you mean 1.**13**.0 (I cant find the changes in 1.3.0) ? :)

**Comment by dahlbyk** [comment]
> Do you mean 1.13.0?

That's what I said 😉

**Pull request merged**